### ISSUE TITLE
HTML tags in administrative workflow view are being evaluated

### ISSUE THREAD

**Issue by firoball** [DSpace/dspace-angular#1495]
**Describe the bug**
First of all, I am not sure whether this is really a bug or more like a feature. 
Since I came across this while doing some XSS attempts on  a local DSpace 7.1.1 setup, I entered some HTML/JS code in the resource submission form.
The sanitizer of Angular appears to detect my attempts and makes them safe. Still, in the administrative workflow the HTML tags are evaluated:

![grafik](https://user-images.githubusercontent.com/4692010/150644929-844c35dc-13ca-424d-97e9-1761015be4a2.png)

**To Reproduce**
Submit new resource, enter
`<img src="https://upload.wikimedia.org/wikipedia/en/thumb/9/9a/Trollface_non-free.png/220px-Trollface_non-free.png" onerror=alert(document.cookie);>`
in any of the text fields.
Hit deposit and check administrative worklow.

**Expected behavior**
HTML tags are stripped entirely or at least are not being evaluated as in the other views.

**Comment by tdonohue** [comment]
Thanks @firoball for the ticket.

* Verified this does NOT occur on Item pages (i.e. `/items/[uuid]`).  On these pages HTML is not evaluated
* However, it strangely DOES occur on any lists of items (seach/browse/mydspace), in either the `dc.title` or  `dc.description.abstract` fields.  Javascript in HTML doesn't appear to execute...but I can get images to load and also links or formatting to work.

This is unexpected behavior. I'll bring this over onto our board to work on.

**Comment by mwoodiupui** [comment]
We have some contributors who write quite elaborate abstracts using XMLUI's "Simple HTML Fragment" feature.  We'll have to clean up these metadata if text-field markup is unimplemented, or implemented in a different way, and the demand will still exist. 

A quick look at [lightweight markup languages](https://en.wikipedia.org/wiki/Lightweight_markup_language) says to me that they are all much too powerful for metadata markup.  If we can eliminate external references (links and images) then this becomes much less problematic.  Maybe an examination of print journals will suggest a small set of essential features.  From a security standpoint, a small language that only deals with formatting would be best.

OTOH I recall that some sites have requested *additional* markup such as MathML.

**Comment by tdonohue** [comment]
NOTE for future work. This needs discussion. The initial ticket here implies that these HTML tags should *not be evaluated*.  But, it seems we *were evaluating some tags* in XMLUI, see https://wiki.lyrasis.org/display/DSDOC7x/Simple+HTML+Fragment+Markup

Therefore, we should discuss whether we can bring that XMLUI feature forward, while also ensuring that XSS (and similar) attacks cannot be achieved (it seems Angular blocks that though, but it may need more testing).

**Comment by mwoodiupui** [comment]
See also #1404

**Comment by mwoodiupui** [comment]
A lot could be accomplished if we could simply avoid normalizing whitespace in these fields.  Then one could have line breaks and paragraph breaks.

**Comment by tdonohue** [comment]
See also #1763 which is loosely related

**Comment by tdonohue** [comment]
Re-verified today.  In DSpace 9, the MyDSpace area will still render embedded HTML (except the `<script>` tag or embedded JS, which is blocked by Angular).  For example, this is a title with an embedded link:
<img width="1336" height="499" alt="Image" src="https://github.com/user-attachments/assets/e2447e47-ab24-402c-9fae-a40b1c8a66ac" />

However, on the Item page itself, any embedded HTML is escaped.. **except** in the Abstract field where it's rendered.

<img width="1228" height="598" alt="Image" src="https://github.com/user-attachments/assets/4449da47-c27b-4eb3-857a-e3a07ae8e54d" />

We need to have more consistent behavior across the application.   I can see an argument for standardizing the behavior to the same as on the Item page, just to avoid confusion.  Either that, or if we plan to render HTML links elsewhere, we may want to warn users that the link it taking them to another site before sending them to that other location.

**Comment by tdonohue** [comment]
As discussed in our [Developer meeting today](https://wiki.lyrasis.org/display/DSPACE/2025-09-11+DSpace+Developers+Meeting), we agreed that HTML in metadata fields should be consistently sanitized across all pages in DSpace, with the behavior of the Item page being the default behavior.

This means that only the Abstract field will render HTML / markdown, because it is [configurable via the `markdown` settings](https://wiki.lyrasis.org/display/DSDOC9x/User+Interface+Configuration#UserInterfaceConfiguration-SettingsforrenderingMarkdown,HTMLandMathJaxinmetadata), allowing sites to turn that rendering on/off.   All other fields (title, etc) should **not render** embedded HTML.  In the future, we could allow additional metadata fields to be configured to enable rendering (similar to the abstract field), but that's not required to complete this ticket.

As an additional enhancement, @artlowel suggested potentially removing embedded HTML from fields shared via OAI-PMH, as that assumes plain text values.  This may mean either dynamically sanitizing fields during OAI indexing, or storing/indexing a "sanitized" version of any field which includes HTML.

As a side note, this HTML rendering behavior is not seen as a security issue because Javascript tags are not allowed throughout DSpace (as they are blocked by Angular) and never rendered.  However, we all agree that allowing for rendering of link tags and similar should be considered a bug for most fields (except the Abstract field, where it will be rendered).   This aligns with behaviors in older versions of DSpace (e.g. XMLUI, as described in comments above).
